tests/test_nonbonded.py

```diff
@@ -515,9 +506,6 @@ def run_nonbonded(precision, potential, x, box, params, lamb, steps=100):
             x = (x.astype(np.float32)).astype(np.float64)
             params = (params.astype(np.float32)).astype(np.float64)
 
-            N = x.shape[0]
-            D = x.shape[1]
```

Comment:
https://github.com/proteneer/timemachine/pull/596#issuecomment-1030172008

Err, `len(x.shape) >= 2`

Reply:
Agreed -- I restored the definitions of array shape constants as comments and added shape assertions in a635a76

Reply:
Nice! This should preserve the previous shape asserting side effect without requiring exceptions to F841.

(Sorry for my confusing comment though: although the code _currently has the effect_ of asserting weaker condition `len(x.shape) >= 2`, the _intent_ is to assert stronger condition `len(x.shape) == 2`)

Reply:
(But it also makes sense to leave as `>=` to minimize any change in behavior during this PR -- no further change requested! 👍 )

Reply:
oops, saw the above too late :) Went ahead and made the assertions more precise in c39b11e, but can revert if we'd rather not change the behavior here

Reply:
Hah, sorry for the flip-flopping -- I prefer the more precise assertions you've now applied!